lib/factor/factorQuadratic.js

```diff
@@ -0,0 +1,127 @@
+'use strict';
+
+const checks = require('../checks');
+const ConstantFactors = require('./ConstantFactors');
+const ChangeTypes = require('../ChangeTypes');
+const flatten = require('../../lib/util/flattenOperands');
+const Negative = require('../Negative');
+const Node = require('../node');
+
+// Given a node, will check if it's in the form of a quadratic equation
+// `ax^2 + bx + c`, and
+// if it is, will factor it using on of the following rules:
+//    - Difference of squares e.g. x^2 - 4 -> (x+2)(x-2)
+//    - Perfect square e.g. x^2 + 2x + 1 -> (x+1)^2
+//    - Sum/product rule e.g. x^2 + 3x + 2 -> (x+1)(x+2)
+//    - TODO: quadratic formula
+function factorQuadratic(node) {
```

Comment:
it'd be nice if this was split into separate function for different cases, but I also see how they all share common variables, so maybe this is best?